File: Submissions/#0582_div3/A/main.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
 
  int n;
  cin >> n;
 
  vector<int> x(n);
  int count = 0;
  
  for (int i = 0; i < n; i++) {
    cin >> x[i];
    if (x[i] % 2 == 0) {
      count++;
    }
  }

  cout << min(count, n - count) << endl;
 
}

File: Submissions/#0582_div3/A/main.java
import java.util.*;

public class Main {

  public static void main (String[] args) {

    Scanner sc = new Scanner(System.in);
    int n = sc.nextInt();

    int[] x = new int[n];
    int count = 0;

    for (int i = 0; i < n; i++) {
      x[i] = sc.nextInt();
      if (x[i] % 2 == 0) {
        count++;
      }
    }

    sc.close();

    System.out.println(Math.min(count, n - count));
    
  }

}